sdk-rs/src/lib.rs

```diff
@@ -0,0 +1,790 @@
+                state: *state_account(),
+                authority: self.wallet.authority(),
+                user: *self.wallet.user(),
+            },
+            self.user,
+            &[],
+            &[],
+        );
+
+        let ix = Instruction {
+            program_id: constants::PROGRAM_ID,
+            accounts,
+            data: InstructionData::data(&drift_program::instruction::CancelOrdersByIds {
+                order_ids,
+            }),
+        };
+        self.ixs.push(ix);
+
+        self
+    }
+
+    /// Modify existing order(s)
+    pub fn modify_orders(mut self, orders: Vec<(u32, OrderParams)>) -> Self {
+        for (order_id, params) in orders {
+            let accounts = build_accounts(
+                self.wallet.context(),
+                drift_program::accounts::PlaceOrder {
+                    state: *state_account(),
+                    authority: self.wallet.authority(),
+                    user: *self.wallet.user(),
+                },
+                self.user,
+                &[],
+                &[],
+            );
+
+            let ix = Instruction {
+                program_id: constants::PROGRAM_ID,
+                accounts,
+                data: InstructionData::data(&drift_program::instruction::ModifyOrder {
+                    order_id: Some(order_id),
+                    modify_order_params: ModifyOrderParams {
+                        direction: Some(params.direction),
+                        base_asset_amount: Some(params.base_asset_amount),
+                        price: Some(params.price),
+                        max_ts: params.max_ts,
+                        oracle_price_offset: params.oracle_price_offset,
+                        auction_duration: params.auction_duration,
+                        auction_end_price: params.auction_end_price,
+                        auction_start_price: params.auction_start_price,
+                        trigger_price: params.trigger_price,
+                        ..Default::default()
+                    },
+                }),
+            };
+            self.ixs.push(ix);
+        }
+
+        self
+    }
+
+    /// Build the transaction ready for signing and sending
+    pub fn build(self) -> Transaction {
+        Transaction::new_with_payer(self.ixs.as_ref(), Some(&self.wallet.authority()))
+    }
+}
+
+/// Builds a set of required accounts from a user's open positions and additional given accounts
+///
+/// `base_accounts` base anchor accounts
+///
+/// `user` Drift user account data
+///
+/// `markets_readable` IDs of markets to include as readable
+///
+/// `markets_writable` ` IDs of markets to include as writable (takes priority over readable)
+///
+/// # Panics
+///  if the user has positions in an unknown market (i.e unsupported by the SDK)
+fn build_accounts(
+    context: Context,
+    base_accounts: impl ToAccountMetas,
+    user: &User,
+    markets_readable: &[MarketId],
+    markets_writable: &[MarketId],
```

Comment:
you might need some way to force perp/spot market always be included: https://github.com/drift-labs/protocol-v2/blob/master/sdk/src/driftClient.ts#L157

seen some race conditions when that doesnt happen where user flips between 0 and open positions, sdk doesnt see account and tx fails because account is missing